### PR TITLE
MMP-196 Create mojos related to packaging lifecycle for mule 3.x

### PR DESCRIPTION
**Pull request by WilliamPinho**

### CODE REVIEW

File: mule-packager/src/main/java/org/mule/tools/api/packager/builder/MulePackageBuilder.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  }
+
+  /**
+   * Creates a mule app package based on the contents of the origin folder, writing them to the destination jar file. The target
+   * file is supposed to have more or less the structure of the example below:
+   * 
+   * <pre>
+   *
+   *
+   * ├── log4j2.xml
```
Comment (damiansima) [inline mule-packager/src/main/java/org/mule/tools/api/packager/builder/MulePackageBuilder.java]: please remove this doc or update it

File: mule-packager/src/test/resources/real-app-target/pom.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.mule.fake</groupId>
+    <artifactId>complete-app</artifactId>
+    <version>1.0.0</version>
+    <name>A Mule Application fake</name>
+
+    <properties>
+        <javaVersion>1.8</javaVersion>
```
Comment (damiansima) [inline mule-packager/src/test/resources/real-app-target/pom.xml]: not major but this branch of code is not meant to work with java 1.8 we'll support up to 1.7 only

File: mule-packager/src/test/java/org/mule/tools/api/packager/MuleProjectFoldersGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    generator = new MuleProjectFoldersGenerator(GROUP_ID, ARTIFACT_ID, PackagingType.MULE);
+    generator.generate(projectBaseFolder.getRoot().toPath());
+
+    assertFolderExist(munitTestMulePath);
+    assertFileExists(munitTestMulePath.resolve(FAKE_FILE_NAME));
+  }
+
+  @Test
+  public void generateMuleSourceFolderAlreadyPresent() throws IOException {
```
Comment (damiansima) [inline mule-packager/src/test/java/org/mule/tools/api/packager/MuleProjectFoldersGeneratorTest.java]: please remove this tests

File: mule-packager/src/test/java/org/mule/tools/api/packager/PackagerTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  public static final String TEST = "test";
+  public static final String TEST_MULE = "test-mule";
+  public static final String MUNIT = "munit";
+
+  public static final String REPOSITORY = "repository";
+
+  public static final String MAVEN = "maven";
+  public static final String MULE_SRC = "mule-src";
+  public static final String MULE_ARTIFACT = "mule-artifact";
```
Comment (damiansima) [inline mule-packager/src/test/java/org/mule/tools/api/packager/PackagerTestUtils.java]: this two should probably go (MULE_SRC, MULE_ARTIFACT)

File: mule-packager/src/main/java/org/mule/tools/api/packager/builder/MulePackageBuilder.java (hunk 1 above)
Comment (WilliamPinho) [reply mule-packager/src/main/java/org/mule/tools/api/packager/builder/MulePackageBuilder.java]: Done

File: mule-packager/src/test/resources/real-app-target/pom.xml (hunk 2 above)
Comment (WilliamPinho) [reply mule-packager/src/test/resources/real-app-target/pom.xml]: The pom is not actually used in the test so I just removed it

File: mule-packager/src/test/java/org/mule/tools/api/packager/MuleProjectFoldersGeneratorTest.java (hunk 3 above)
Comment (WilliamPinho) [reply mule-packager/src/test/java/org/mule/tools/api/packager/MuleProjectFoldersGeneratorTest.java]: Done

File: mule-packager/src/test/java/org/mule/tools/api/packager/PackagerTestUtils.java (hunk 4 above)
Comment (WilliamPinho) [reply mule-packager/src/test/java/org/mule/tools/api/packager/PackagerTestUtils.java]: Done

**Pull request merged**